### PR TITLE
Update error handling in viz container

### LINKED ISSUE #161: [BUG] Show user friendly errors in Panel Visualizations

### PR DESCRIPTION
**Pull request by ps48**
Signed-off-by: Shenoy Pratik <sgguruda@amazon.com>

### Description
Update error handling in viz container to show user friendly errors in Panel Visualizations. 

### Issues Resolved
#161 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by mengweieric — APPROVED** [review]

File: public/components/custom_panels/panel_modules/visualization_flyout/visualization_flyout.tsx [hunk 1]
```diff
@@ -325,12 +326,12 @@ export const VisaulizationFlyout = ({
...
               <EuiLoadingChart size="xl" mono className="visualization-loading-chart-preview" />
-            ) : isPreviewError !== '' ? (
-              <div className="visualization-error-div-preview">
+            ) : !_.isEmpty(isPreviewError) ? (
+              <div className="visualization-error-div">
                 <EuiIcon type="alert" color="danger" size="s" />
                 <EuiSpacer size="s" />
                 <EuiText size="s">
-                  <p>{isPreviewError}</p>
+                  <p>{isPreviewError.error}</p>
```
Comment (derek-ho) [inline public/components/custom_panels/panel_modules/visualization_flyout/visualization_flyout.tsx:334]: is this supposed to be errorMessage or errorDetails?

File: public/components/custom_panels/panel_modules/visualization_container/visualization_container.scss [hunk 2]
```diff
@@ -26,6 +26,12 @@
   height: 90%;
   overflow: auto;
   text-align: center;
+
+  button {
+    height: 26px !important;
```
[the remark this answers is not in the document]
Comment (derek-ho) [reply public/components/custom_panels/panel_modules/visualization_container/visualization_container.scss:31]: Can you explain what this is trying to achieve? Not too familiar with css in general

File: public/components/custom_panels/panel_modules/visualization_container/visualization_container.tsx [hunk 3]
```diff
@@ -99,42 +101,67 @@ export const VisualizationContainer = ({
...
   const [visualizationData, setVisualizationData] = useState<Plotly.Data[]>([]);
   const [isLoading, setIsLoading] = useState(true);
-  const [isError, setIsError] = useState('');
+  const [isError, setIsError] = useState({} as VizContainerError);
   const onActionsMenuClick = () => setIsPopoverOpen((currPopoverOpen) => !currPopoverOpen);
   const closeActionsMenu = () => setIsPopoverOpen(false);
 
   const [isModalVisible, setIsModalVisible] = useState(false);
+  const [modalContent, setModalContent] = useState(<></>);
+  // let modal;
```
Comment (derek-ho) [inline public/components/custom_panels/panel_modules/visualization_container/visualization_container.tsx:110]: remove comment

**Review by derek-ho — APPROVED** [review]

File: public/components/custom_panels/panel_modules/visualization_container/visualization_container.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (ps48) [reply public/components/custom_panels/panel_modules/visualization_container/visualization_container.scss:31]: `important` overrides al the other CSS declarations for that particular element. Usually it is avoided as it becomes difficult to debug later on. But, in our case with EUI components having their own CSS it gets difficult to override them with a class name.

File: public/components/custom_panels/panel_modules/visualization_container/visualization_container.tsx (hunk 3 above)
Comment (ps48) [reply public/components/custom_panels/panel_modules/visualization_container/visualization_container.tsx]: fixed here: https://github.com/opensearch-project/dashboards-observability/pull/247/commits/c46c072d78eae320e585105418d70cedd86e7514

File: public/components/custom_panels/panel_modules/visualization_flyout/visualization_flyout.tsx (hunk 1 above)
Comment (ps48) [reply public/components/custom_panels/panel_modules/visualization_flyout/visualization_flyout.tsx]: Thanks for the catch missed this. Updated : https://github.com/opensearch-project/dashboards-observability/pull/247/commits/c46c072d78eae320e585105418d70cedd86e7514

**Pull request merged**